apps/demo/src/H5GroveApp.tsx

```diff
@@ -3,8 +3,8 @@ import { useLocation } from 'react-router-dom';
 
 import { getFeedbackURL } from './utils';
 
-const URL = process.env.REACT_APP_H5GROVE_URL || '';
-const FILEPATH = process.env.REACT_APP_H5GROVE_FALLBACK_FILEPATH || '';
+const URL = import.meta.env.VITE_H5GROVE_URL as string;
+const FILEPATH = import.meta.env.VITE_H5GROVE_FALLBACK_FILEPATH as string;
```

Comment:
Note the change from `process.env` to `import.meta.env`. I've also changed the empty string fallbacks into type casts because Vite types env vars as `string | boolean | undefined` -- the `boolean` part is weird, as in my tests I have not been able to get an env var parsed as boolean... See https://github.com/vitejs/vite/discussions/6637

Reply:
Isn't the type cast `as string` wrong if the variable is not set ? It will be `undefined` right ?

Reply:
All the variables are set as long as they're listed in `.env` - i.e. `VITE_H5GROVE_URL=` assigns an empty string to the variable. Also, it's slightly better to fail hardly because of `undefined` rather than silently because of an empty string.

Ideally, we'd want to assert that all the env vars respect some sort of schema (e.g. either `'true'` or `''` for `VITE_DIST`), but I haven't found a way to do this cleanly at runtime. It works with `assertStr` or equivalent, but only when the guard is invoked inside the function in which the env var is used. I can't assert all the en vars globally without writing unnecessarily complex code to access the asserted variables (in my opinion).

If Vite had a plugin of sorts, it'd be amazing. There's https://github.com/ElMassimo/vite-plugin-environment but I'm not a fan of the API (setting an env var's default value to `undefined` to say that it's required...) So I went with the status quo... 😄